integration/test/watcher/configmap/configmap_test.go

```diff
@@ -0,0 +1,355 @@
+
+		appCatalogCR := &v1alpha1.AppCatalog{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: key.DefaultCatalogName(),
+				Labels: map[string]string{
+					label.AppOperatorVersion: key.UniqueAppVersion(),
+				},
+			},
+			Spec: v1alpha1.AppCatalogSpec{
+				Config: v1alpha1.AppCatalogSpecConfig{
+					ConfigMap: v1alpha1.AppCatalogSpecConfigConfigMap{
+						Name:      key.AppCatalogConfigMapName(),
+						Namespace: key.Namespace(),
+					},
+				},
+				Description: key.DefaultCatalogName(),
+				Title:       key.DefaultCatalogName(),
+				Storage: v1alpha1.AppCatalogSpecStorage{
+					Type: "helm",
+					URL:  key.DefaultCatalogStorageURL(),
+				},
+			},
+		}
+		_, err = config.K8sClients.G8sClient().ApplicationV1alpha1().AppCatalogs().Create(ctx, appCatalogCR, metav1.CreateOptions{})
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("created %#q appcatalog cr", key.DefaultCatalogName()))
+	}
+
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("creating configmap %#q in namespace %#q", key.UserConfigMapName(), key.Namespace()))
+
+		cm := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      key.UserConfigMapName(),
+				Namespace: key.Namespace(),
+			},
+			Data: map[string]string{
+				"values": "",
+			},
+		}
+
+		_, err := config.K8sClients.K8sClient().CoreV1().ConfigMaps(key.Namespace()).Create(ctx, cm, metav1.CreateOptions{})
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("created configmap %#q in namespace %#q", key.UserConfigMapName(), key.Namespace()))
+	}
+
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("creating %#q app cr", key.TestAppReleaseName()))
+
+		appCR := &v1alpha1.App{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      key.TestAppReleaseName(),
+				Namespace: key.Namespace(),
+				Labels: map[string]string{
+					label.AppOperatorVersion: key.UniqueAppVersion(),
+				},
+			},
+			Spec: v1alpha1.AppSpec{
+				Catalog: key.DefaultCatalogName(),
+				KubeConfig: v1alpha1.AppSpecKubeConfig{
+					InCluster: true,
+				},
+				Name:      key.TestAppReleaseName(),
+				Namespace: key.Namespace(),
+				UserConfig: v1alpha1.AppSpecUserConfig{
+					ConfigMap: v1alpha1.AppSpecUserConfigConfigMap{
+						Name:      key.UserConfigMapName(),
+						Namespace: key.Namespace(),
+					},
+				},
+				Version: "0.1.0",
+			},
+		}
+
+		_, err = config.K8sClients.G8sClient().ApplicationV1alpha1().Apps(key.Namespace()).Create(ctx, appCR, metav1.CreateOptions{})
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("creating %#q app cr", key.TestAppReleaseName()))
+	}
+
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "waiting until user configmap being labelled")
```

Comment:
```suggestion
		config.Logger.LogCtx(ctx, "level", "debug", "message", "waiting until user configmap is labelled")
```

Thanks for using 🇬🇧 spelling! ❤️